File: src/Features/VisualBasic/Portable/Structure/Providers/FieldDeclarationStructureProvider.vb
﻿' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports System.Collections.Immutable
Imports System.Threading
Imports Microsoft.CodeAnalysis.Options
Imports Microsoft.CodeAnalysis.PooledObjects
Imports Microsoft.CodeAnalysis.Structure
Imports Microsoft.CodeAnalysis.VisualBasic.Syntax

Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
    Friend Class FieldDeclarationStructureProvider
        Inherits AbstractSyntaxNodeStructureProvider(Of FieldDeclarationSyntax)

        Protected Overrides Sub CollectBlockSpans(fieldDeclaration As FieldDeclarationSyntax,
                                                  spans As ArrayBuilder(Of BlockSpan),
                                                  options As OptionSet,
                                                  cancellationToken As CancellationToken)
            CollectCommentsRegions(fieldDeclaration, spans)
        End Sub
    End Class
End Namespace

File: src/Features/CSharp/Portable/Structure/Providers/MetadataAsSource/FieldDeclarationStructureProvider.cs
﻿// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace Microsoft.CodeAnalysis.CSharp.Structure.MetadataAsSource
{
    internal class MetadataFieldDeclarationStructureProvider : AbstractMetadataAsSourceStructureProvider<FieldDeclarationSyntax>
    {
        protected override SyntaxToken GetEndToken(FieldDeclarationSyntax node)
        {
            return node.Modifiers.Count > 0
                    ? node.Modifiers.First()
                    : node.Declaration.GetFirstToken();
        }
    }
}
